tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/Process/ProcessHelper.cs

```diff
@@ -84,22 +129,39 @@ public ProcessResult RunProcess(string command, string[] args, string workingDir
                     }
                 };
 
-                logger.LogDebug($"Running command: {command} {string.Join(" ", args)} in {workingDirectory}");
+                logger.LogDebug("Running command: {command} {args} in {workingDirectory}", command, string.Join(" ", args), workingDirectory);
 
                 process.Start();
                 process.BeginOutputReadLine();
                 process.BeginErrorReadLine();
 
-                process.WaitForExit(processTimeout);
-                exitCode = process.ExitCode;
+                try
+                {
+                    await process.WaitForExitAsync(linkedCts.Token);
+                }
+                catch (OperationCanceledException) when (timeoutCts.IsCancellationRequested)
+                {
+                    output.AppendLine();
+                    output.AppendLine($"Process '{command}' timed out after {timeout.TotalMilliseconds}ms");
+                    logger.LogError("Process '{command}' timed out after {timeout}ms", command, timeout.TotalMilliseconds);
+                    return new ProcessResult { Output = output.ToString(), ExitCode = 124 };
```

Comment:
We should avoid overloading ExitCode like this. If our worry is that it's difficult to tell if a ProcessResult represents a failure we can add a method, like `bool Failed()`, so people aren't making assumptions based on the exit code.

The other option is just to let the exception out. We could create a subclass and just add in Output as one of the fields.

Reply:
There is no original process exit code in this timeout case. Or are you saying we should just always have exit code be 1?